### PR TITLE
Create DeleteSuggestionCommand and DeleteSuggestionCommandTest

### LINKED ISSUE #238: Create DeleteSuggestionCommand and DeleteSuggestionCommandTest
As `DeleteSuggestionCommand` has the same logic as `OpenSuggestionCommand`, the code base is similar.

`DeleteSuggestionCommand` stores a `responseText` of "Delete a note" and a list of `SuggestionItem` of possible paths into `Model`.

### PR DESCRIPTION
**Pull request by johannagwan**
Closes #238

### CODE REVIEW

File: src/main/java/com/notably/logic/suggestion/commands/DeleteSuggestionCommand.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+    private List<SuggestionItem> getSuggestions(List<AbsolutePath> possiblePaths, Model model) {
+        Objects.requireNonNull(possiblePaths);
+        Objects.requireNonNull(model);
+
+        return possiblePaths.stream()
+                .map(path -> {
+                    String displayText = path.getStringRepresentation();
+                    Runnable action = () -> {
+                        model.setInput(displayText);
```
Comment (kevinputera) [inline src/main/java/com/notably/logic/suggestion/commands/DeleteSuggestionCommand.java]: Found a bug here (I missed this in your `OpenSuggestionCommand` PR, please fix it!). This should be `model.setInput(COMMAND_WORD + “ “ + PREFIX_TITLE + “ “ + displayText)` right?

File: src/test/java/com/notably/logic/suggestion/commands/DeleteSuggestionCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    @Test
+    public void execute_nullModel_throwsNullPointerException() {
+        DeleteSuggestionCommand deleteSuggestionCommand = new DeleteSuggestionCommand(toRoot);
+        assertThrows(NullPointerException.class, () -> deleteSuggestionCommand.execute(null));
+    }
+
+    @Test
+    public void execute_displayText() {
```
Comment (kevinputera) [inline src/test/java/com/notably/logic/suggestion/commands/DeleteSuggestionCommandTest.java]: Actually this should be called `execute_generateResponseCorrectly` or something else that is more general? `displayText` is just a subset of what you’re testing here

**Review by kevinputera — CHANGES_REQUESTED** [review]
There’s a bug on the list suggestion part! Other than that things looks good!

File: src/main/java/com/notably/logic/suggestion/commands/DeleteSuggestionCommand.java (hunk 1 above)
Comment (johannagwan) [reply src/main/java/com/notably/logic/suggestion/commands/DeleteSuggestionCommand.java]: Oh yes sorry for the careless mistake!!

File: src/test/java/com/notably/logic/suggestion/commands/DeleteSuggestionCommandTest.java (hunk 2 above)
Comment (johannagwan) [reply src/test/java/com/notably/logic/suggestion/commands/DeleteSuggestionCommandTest.java]: alright!

**Comment by johannagwan** [comment]
@kevinputera for the `PREFIX_TITLE`, I will still import from `com.notably.logic.parser.CliSyntax.PREFIX_TITLE`. This will be changed after I reorganize the packages.

**Review by kevinputera — APPROVED** [review]
LGTM

**Pull request merged**